gamestonk_terminal/cryptocurrency/coinpaprika/coinpaprika_view.py

```diff
@@ -0,0 +1,1541 @@
+            "confidence",
+            "volume_24h",
+            "volume_7d",
+            "volume_30d",
+            "sessions_per_month",
+        ],
+    )
+
+    parser.add_argument(
+        "--descend",
+        action="store_false",
+        help="Flag to sort in descending order (lowest first)",
+        dest="descend",
+        default=True,
+    )
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+
+        df = paprika.get_list_of_exchanges(quotes=ns_parser.vs).sort_values(
+            by=ns_parser.sortby, ascending=ns_parser.descend
+        )
+
+        if df.empty:
+            return
+
+        df = df.applymap(lambda x: long_number_format_with_type_check(x))
+        print("")
+        print(f"Displaying data vs {ns_parser.vs}")
+        print(
+            tabulate(
+                df.head(ns_parser.top),
+                headers=df.columns,
+                floatfmt=".2f",
+                showindex=False,
+                tablefmt="fancy_grid",
+            )
+        )
+        print("")
+
+    except Exception as e:
+        print(e)
+        print("")
+
+
+def search(other_args: List[str]):
+    """Search in CoinPaprika
+
+    Parameters
+    ----------
+    other_args: List[str]
+        Arguments to pass to argparse
+
+    """
+    parser = argparse.ArgumentParser(
+        prog="search",
+        add_help=False,
+        description="""Search over CoinPaprika API
+        You can display only top N number of results with --top parameter.
+        You can sort data by id, name , category --sort parameter and also with --descend flag to sort descending.
+        To choose category in which you are searching for use --cat/-c parameter. Available categories:
+        currencies|exchanges|icos|people|tags|all
+        Displays:
+            id, name, category
+        """,
+    )
+
+    parser.add_argument(
+        "-q", "--query", help="phrase for search", dest="query", type=str, required=True
+    )
+
+    parser.add_argument(
+        "-c",
+        "--cat",
+        help="Categories to search: currencies|exchanges|icos|people|tags|all. Default: all",
+        dest="category",
+        default="all",
+        type=str,
+        choices=[
+            "currencies",
+            "exchanges",
+            "icos",
+            "people",
+            "tags",
+            "all",
+        ],
+    )
+
+    parser.add_argument(
+        "-t", "--top", default=20, dest="top", help="Limit of records", type=int
+    )
+
+    parser.add_argument(
+        "-s",
+        "--sort",
+        dest="sortby",
+        type=str,
+        help="Sort by given column. Default: id",
+        default="id",
+        choices=["category", "id", "name"],
+    )
+
+    parser.add_argument(
+        "--descend",
+        action="store_false",
+        help="Flag to sort in descending order (lowest first)",
+        dest="descend",
+        default=True,
+    )
+
+    try:
```

Comment:
In commands where there's a required argument, as is the case of this one. You can't see the Help, e.g.
```
(✨) (crypto)>(cp)> search -h
usage: search -q QUERY [-c {currencies,exchanges,icos,people,tags,all}] [-t TOP] [-s {category,id,name}] [--descend] [-h]
search: error: the following arguments are required: -q/--query
The command selected doesn't exist
```

The best way I found to overcome this is by doing:
```
parser.add_argument(
         "-q", 
         "--query", 
         help="phrase for search", 
         dest="query", 
         type=str, 
         required="-h" not in other_args,
     )
```
Note the `required` parameter.

Reply:
Wow, very nice solution. I was thinking about how to solve this and I couldn't find any solution. Yours is great!